rust/core-lib/src/tabs.rs

```diff
@@ -807,19 +815,25 @@ impl CoreState {
 
     /// Handles changes in theme files.
     #[cfg(feature = "notify")]
-    fn handle_themes_fs_event(&mut self, event: DebouncedEvent) {
-        use self::DebouncedEvent::*;
-        match event {
-            Create(ref path) | Write(ref path) => self.load_theme_file(path),
+    fn handle_themes_fs_event(&mut self, event: Event) {
+        use notify::event::*;
+        match event.kind {
+            EventKind::Create(CreateKind::Any) | EventKind::Modify(ModifyKind::Any) => {
+                self.load_theme_file(&event.paths[0])
+            }
             // the way FSEvents on macOS work, we want to verify that this path
             // has actually be removed before we do anything.
-            NoticeRemove(ref path) | Remove(ref path) if !path.exists() => self.remove_theme(path),
-            Rename(ref old, ref new) => {
+            EventKind::Remove(RemoveKind::Any) if !event.paths[0].exists() => {
+                self.remove_theme(&event.paths[0]);
+            }
+            EventKind::Modify(ModifyKind::Name(RenameMode::Both)) => {
+                let old = &event.paths[0];
+                let new = &event.paths[1];
                 self.remove_theme(old);
                 self.load_theme_file(new);
             }
-            Chmod(ref path) | Remove(ref path) => {
-                self.style_map.borrow_mut().sync_dir(path.parent())
+            EventKind::Modify(ModifyKind::Metadata(MetadataKind::Any)) => {
```

Comment:
Shouldn't `EventKind::Remove` be matched here too?

Reply:
It's already been matched on line 826.

Reply:
it has only been matched if thee path does not exist; we need to handle the case where the path does exist, see my earlier comment. (This is confusing, I understand).

Reply:
Pattern guard here as well. Making this change.